receiver/otlpreceiver/otlp.go

```diff
@@ -145,13 +145,13 @@ func (r *otlpReceiver) startHTTPServer(host component.Host) error {
 	}
 
 	var err error
-	if r.serverHTTP, err = r.cfg.HTTP.ToServer(host, r.settings.TelemetrySettings, httpMux, confighttp.WithErrorHandler(errorHandler)); err != nil {
+	if r.serverHTTP, err = r.cfg.HTTP.ToServerContext(context.Background(), host, r.settings.TelemetrySettings, httpMux, confighttp.WithErrorHandler(errorHandler)); err != nil {
```

Comment:
Does it make sense to update `startHTTPServer` to receive the context and pass it in?

Reply:
What would you have me do ? @codeboten

Reply:
i'm suggesting updating the method signature for `startHTTPServer` (at line 119 and then all the calls to it) to also receive a `context.Context` argument that can be passed in to line 148

If you look at line 175, the `Start` method could pass the `ctx` into `startHTTPServer`. Does that make sense?

Reply:
Yes i understand
Will add context to the method